vispy/visuals/colorbar.py

```diff
@@ -604,6 +604,29 @@ def label(self, label):
         self._label = label
         self._update()
```

Comment:
I'm not sure why it took me this wrong to realize this, but why can't all of this be done with an `if` statement for the regular `label` property? I know I've expressed my dislike of `label_str` as a property and I didn't see a way around it before, but why not do:

```
@label.setter
def label(self, label):
    if isinstance(label, str):
        self._label.text = label
    else:
        self._label = label
        self._update()
```

I'm not sure if the `_update` needs to be done in both cases since the TextVisual would call its own update. With this change the `label_str` property could be removed. Additionally, have you actually tested this? When does the `ColorbarVisual._label_str` get used? It doesn't seem to show up in the `_update` function.

Thoughts?

Reply:
@djhoese :+1:

Reply:
I was using this code here https://github.com/macrocosme/shwirl/blob/master/shwirl/extern/vispy/scene/widgets/colorbar.py without a fuss. But I should indeed sit down and properly test this. I'm travelling so it may be a few days.

Reply:
As to merging the two with an if, I like it! 👍